### PR TITLE
introduce offset to all queries, adds tests for it

### PR DESCRIPTION
**Pull request by tsmith023**

### CODE REVIEW

File: integration/test_collection.py [hunk 1]
```diff
@@ -605,20 +606,82 @@ def test_search_hybrid_only_vector(collection_factory: CollectionFactory) -> Non
     assert len(objs) == 2
 
 
-@pytest.mark.parametrize("limit", [1, 5])
-def test_search_limit(collection_factory: CollectionFactory, limit: int) -> None:
+@pytest.mark.parametrize("limit", [1, 2])
+def test_hybrid_limit(collection_factory: CollectionFactory, limit: int) -> None:
     collection = collection_factory(
-        properties=[Property(name="Name", data_type=DataType.TEXT)],
+        properties=[Property(name="Name", data_type=DataType.TEXT, tokenization=Tokenization.WORD)],
```
Comment (dirkkul) [inline integration/test_collection.py:612]: why tokenozation here?

File: integration/test_collection.py [hunk 2]
```diff
@@ -605,20 +606,82 @@ def test_search_hybrid_only_vector(collection_factory: CollectionFactory) -> Non
...
+            {"Name": "another"},
+            {"Name": "test"},
+        ]
+    )
+    assert res.has_errors is False
+    assert len(collection.query.hybrid(query="test", alpha=0, limit=limit).objects) == limit
+
+
+@pytest.mark.parametrize("offset,expected", [(0, 2), (1, 1), (2, 0)])
+def test_hybrid_offset(collection_factory: CollectionFactory, offset: int, expected: int) -> None:
```
Comment (dirkkul) [inline integration/test_collection.py:628]: maybe move all of these to test_offset? but only if you want to

File: weaviate/collections/queries/near_media/query.py [hunk 3]
```diff
@@ -58,7 +59,9 @@ def near_media(
             `distance`
                 The maximum distance to search. If not specified, the default distance specified by the server is used.
             `limit`
-                The maximum number of results to return. If not specified, the default limit specified by the server is returned.
+                The maximum number of results to return. If not specified, the default limit specified by the server is returned.\
```
Comment (dirkkul) [inline weaviate/collections/queries/near_media/query.py:62]: remove char at end

File: integration/test_collection.py (hunk 1 above)
Comment (tsmith023) [reply integration/test_collection.py:612]: Ah, relic from before I set `alpha=0` and forgot to remove

File: integration/test_collection.py (hunk 2 above)
Comment (tsmith023) [reply integration/test_collection.py:628]: I'd prefer them separate to decouple the different collection configs involved in each one

File: weaviate/collections/queries/near_media/query.py (hunk 3 above)
Comment (tsmith023) [reply weaviate/collections/queries/near_media/query.py:62]: great catch!

**Pull request merged**